src/drc/api/viewsets.py

```diff
@@ -95,10 +101,42 @@ class EnkelvoudigInformatieObjectViewSet(NotificationViewSetMixin,
         'destroy': SCOPE_DOCUMENTEN_ALLES_VERWIJDEREN,
         'update': SCOPE_DOCUMENTEN_BIJWERKEN,
         'partial_update': SCOPE_DOCUMENTEN_BIJWERKEN,
+        'lock': SCOPE_DOCUMENTEN_LOCK,
+        'unlock': SCOPE_DOCUMENTEN_LOCK | SCOPE_DOCUMENTEN_GEFORCEERD_UNLOCK
     }
     notifications_kanaal = KANAAL_DOCUMENTEN
     audit = AUDIT_DRC
 
+    @swagger_auto_schema(request_body=LockEnkelvoudigInformatieObjectSerializer,
+                         responses={200: LockEnkelvoudigInformatieObjectSerializer})
+    @action(detail=True, methods=['post'])
+    def lock(self, request, *args, **kwargs):
+        eio = self.get_object()
+        lock_serializer = LockEnkelvoudigInformatieObjectSerializer(eio, data=request.data)
+        lock_serializer.is_valid(raise_exception=True)
+        lock_serializer.save()
+        return Response(lock_serializer.data)
+
+    @swagger_auto_schema(request_body=UnlockEnkelvoudigInformatieObjectSerializer,
+                         responses={200: UnlockEnkelvoudigInformatieObjectSerializer})
+    @action(detail=True, methods=['post'])
+    def unlock(self, request, *args, **kwargs):
+        eio = self.get_object()
+        # check if it's a force unlock by administrator
+        force_unlock = False
+        if self.request.jwt_auth.has_auth(
+            scopes=SCOPE_DOCUMENTEN_GEFORCEERD_UNLOCK,
+            informatieobjecttype=eio.informatieobjecttype,
+            vertrouwelijkheidaanduiding=eio.vertrouwelijkheidaanduiding
+        ):
+            force_unlock = True
+        unlock_serializer = UnlockEnkelvoudigInformatieObjectSerializer(eio, data=request.data)
+        unlock_serializer.context['force_unlock'] = force_unlock
+
+        unlock_serializer.is_valid(raise_exception=True)
+        unlock_serializer.save()
+        return Response(status=status.HTTP_200_OK)
```

Comment:
since there's no data returned, maybe `status.HTTP_204_EMPTY` is more correct

Reply:
status is changed from 200 to 204